### PR TITLE
HPCC-28746 Improve reporting of Maximum Packet Length Exceeded errors

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Used a trivial bit of ECL

### CODE REVIEW

File: roxie/ccd/ccdqueue.cpp [hunk 1]
```diff
@@ -2097,10 +2122,7 @@ class RoxieThrottledPacketSender : public Thread
             break;
         }
         if (length > maxPacketSize)
-        {
-            StringBuffer s;
-            throw MakeStringException(ROXIE_PACKET_ERROR, "Maximum packet length %d exceeded sending packet %s", maxPacketSize, header.toString(s).str());
-        }
+            throwPacketTooLarge(x, maxPacketSize);
```
Comment (mckellygit) [inline roxie/ccd/ccdqueue.cpp:2125]: Is it just my tab set up or is indentation off here ?

**Review by mckellygit — COMMENTED** [review]
A few minor questions/comments

File: roxie/ccd/ccdqueue.cpp [hunk 2]
```diff
@@ -1988,6 +1988,31 @@ class RoxieReceiverBase : implements IRoxieOutputQueueManager, public CInterface
...
+        const byte *traceInfo = x->queryTraceInfo();
+        unsigned char loggingFlags = *traceInfo;
+        if (loggingFlags & LOGGING_FLAGSPRESENT) // should always be true.... but this flag is handy to avoid flags byte ever being NULL
+        {
+            traceInfo++;
+            traceLength--;
+            if (loggingFlags & LOGGING_TRACELEVELSET)
+            {
+                traceInfo++;
+                traceLength--;
```
Comment (mckellyln) [inline roxie/ccd/ccdqueue.cpp:2006]: Could traceLength ever be 0 before this decrement ?

File: roxie/ccd/ccdqueue.cpp [hunk 3]
```diff
@@ -2097,10 +2122,7 @@ class RoxieThrottledPacketSender : public Thread
             break;
         }
         if (length > maxPacketSize)
```
Comment (mckellyln) [inline roxie/ccd/ccdqueue.cpp:2124]: Probably not a part of this PR, but should we subtract a few bytes from maxPacketSize to allow for IP+UDP header ?

**Review by mckellyln — COMMENTED** [review]
Just minor comment

File: roxie/ccd/ccdqueue.cpp (hunk 2 above)
Comment (richardkchapman) [reply roxie/ccd/ccdqueue.cpp:2006]: Only if packets were corrupt. There's similar code at the agent end interpreting the received data...

File: roxie/ccd/ccdqueue.cpp (hunk 3 above)
Comment (richardkchapman) [reply roxie/ccd/ccdqueue.cpp:2124]: Possibly - sounds like a new Jira though

Comment (mckellyln) [reply roxie/ccd/ccdqueue.cpp:2124]: I agree, new JIRA

**Review by mckellyln — APPROVED** [review]
Looks good.

**Review by ghalliday — APPROVED** [review]
@richardkchapman looks good, but can you target 8.10.x instead - it would be userful earlier rather than later

**Comment by richardkchapman** [comment]
@ghalliday Rebased and retargeted

**Pull request merged**